### PR TITLE
chore: Release Zebra 1.0.0-beta.15

### LINKED ISSUE #5011: Tag Zebra 1.0.0-beta.15 release
## Motivation

Every few weeks we tag a Zebra release.

### PR DESCRIPTION
**Pull request by teor2345**
Closes #5011

### Which Crates to Increment

To check if any of the top-level crates need version increments:
1. Go to the zebra GitHub code page: https://github.com/ZcashFoundation/zebra
2. Check if the last commit was a Zebra version bump, or something else

<details>
   
Alternatively you can:
- Use the github compare tool and check the `main` branch against the last tag ([Example](https://github.com/ZcashFoundation/zebra/compare/v1.0.0-alpha.15...main))
- Use `git diff --stat <previous-release-tag> origin/main`

</details>
   
Once you know which crates have changed:
- [x] Increment the crates that have new commits since the last version update
- [x] Increment any crates that depend on crates that have changed
- [x] Keep a list of the crates that haven't been incremented, to include in the PR

### How to Increment Versions

Zebra follows [semantic versioning](https://semver.org).

Semantic versions look like: MAJOR`.`MINOR`.`PATCH[`-`TAG`.`PRE-RELEASE]

### Reviewing Version Bumps

Check for missed changes by going to:
`https://github.com/ZcashFoundation/zebra/tree/<commit-hash>/`
Where `<commit-hash>` is the hash of the last commit in the version bump PR.

If any Zebra or Tower crates have commit messages that are **not** a version bump, we have missed an update.
Also check for crates that depend on crates that have changed. They should get a version bump as well.
   
### Version Locations

Once you know which versions you want to increment, you can find them in the:
- [x] zebra* `Cargo.toml`s
- [x] tower-* `Cargo.toml`s
- [x] `zebra-network` protocol user agent: https://github.com/ZcashFoundation/zebra/blob/main/zebra-network/src/constants.rs
- [x] `README.md`
- [x] `book/src/user/install.md`
- [x] `Cargo.lock`: automatically generated by `cargo build`

#### Version Tooling

You can use `fastmod` to interactively find and replace versions.

For example, you can do something like:
```
fastmod --extensions rs,toml,md --fixed-strings '1.0.0-beta.11' '1.0.0-beta.12'
fastmod --extensions rs,toml,md --fixed-strings '0.2.26' '0.2.27' tower-batch tower-fallback
```

If you use `fastmod`, don't update versions in `CHANGELOG.md`.

## README

We should update the README to:
- [x] Remove any "Known Issues" that have been fixed
- [x] Update the "Build and Run Instructions" with any new dependencies.
      Check for changes in the `Dockerfile` since the last tag: `git diff <previous-release-tag> docker/Dockerfile`.

## Checkpoints

With every release and for performance reasons, we want to update the zebra checkpoints. More information on how to do this can be found in [the zebra-checkpoints README](https://github.com/ZcashFoundation/zebra/blob/main/zebra-consensus/src/checkpoint/README.md).

To do this you will need a synchronized `zcashd` node. You can request help from other zebra team members to submit this PR if you can't make it yourself at the moment of the release.

## Change Log

**Important**: Any merge into `main` deletes any edits to the draft changelog.
Once you are ready to tag a release, copy the draft changelog into `CHANGELOG.md`.

We follow the [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) format.

We use [the Release Drafter workflow](https://github.com/marketplace/actions/release-drafter) to automatically create a [draft changelog](https://github.com/ZcashFoundation/zebra/releases).

To create the final change log:
- [x] Copy the draft changelog into `CHANGELOG.md`
- [x] Delete any trivial changes. Keep the list of those, to include in the PR
- [x] Combine duplicate changes
- [x] Edit change descriptions so they are consistent, and make sense to non-developers
- [x] Check the category for each change
  - Prefer the "Fix" category if you're not sure

<details>
   
#### Change Categories

From "Keep a Changelog":
* `Added` for new features.
* `Changed` for changes in existing functionality.
* `Deprecated` for soon-to-be removed features.
* `Removed` for now removed features.
* `Fixed` for any bug fixes.
* `Security` in case of vulnerabilities.

</details>

### Create the Release PR

After you have the version increments, the updated checkpoints and the updated changelog:

- [x] Make sure the PR with the new checkpoint hashes is already merged.
- [x] Push the version increments and the updated changelog into a branch
      (name suggestion, example: `v1.0.0-alpha.0-release`)
- [x] Create a release PR by adding `&template=release-checklist.md` to the
      comparing url ([Example](https://github.com/ZcashFoundation/zebra/compare/v1.0.0-alpha.0-release?expand=1&template=release-checklist.md)).
  - [x] Add the list of deleted changelog entries as a comment to make reviewing easier.
  - [x] Also add the list of not-bumped crates as a comment (can use the same comment as the previous one).
- [x] Turn on [Merge Freeze](https://www.mergefreeze.com/installations/3676/branches).
- [x] Once the PR is ready to be merged, unfreeze it [here](https://www.mergefreeze.com/installations/3676/branches).
      Do not unfreeze the whole repository.

### Create the Release

- [x] Once the PR has been merged,
      create a new release using the draft release as a base,
      by clicking the Edit icon in the [draft release](https://github.com/ZcashFoundation/zebra/releases)
- [x] Set the tag name to the version tag,
      for example: `v1.0.0-alpha.0`
- [x] Set the release to target the `main` branch
- [x] Set the release title to `Zebra ` followed by the version tag,
      for example: `Zebra 1.0.0-alpha.0`
- [x] Replace the prepopulated draft changelog in the release description by the final
      changelog you created; starting just _after_ the title `## [Zebra ...` of
      the current version being released, and ending just _before_ the title of
      the previous release.
- [x] Mark the release as 'pre-release', until it has been built and tested
- [x] Publish the pre-release to GitHub using "Publish Release"

## Build and Binary Testing

- [x] After tagging the release, test that the exact `cargo install` command in `README.md` works
      (`--git` behaves a bit differently to `--path`)
- [x] Test that the newly built Zebra starts correctly, by running `~/.cargo/bin/zebrad`
- [x] Wait until the [Docker binaries have been built on `main`](https://github.com/ZcashFoundation/zebra/actions/workflows/continous-integration-docker.yml), and the quick tests have passed.
      (You can ignore the full sync and `lightwalletd` tests, because they take about a day to run.)
- [x] Publish the release to GitHub by disabling 'pre-release', then clicking "Publish Release"
- [x] Wait until [the Docker images have been published](https://github.com/ZcashFoundation/zebra/actions/workflows/release-binaries.yml)
- [x] Turn off [Merge Freeze](https://www.mergefreeze.com/installations/3676/branches) for the whole repository

If the build fails after tagging:
1. Fix the build
2. Increment versions again, following these instructions from the start
3. Update `README.md` with a **new** git tag
4. Update `CHANGELOG.md` with details about the fix
5. Tag a **new** release

### CODE REVIEW

**Comment by teor2345** [comment]
Here's how I made the release PR:
- All crates needed version bumps
- I didn't update the checkpoints, they were just updated last week

I combined or deleted these PRs in the draft changelog:
- build(deps): bump tokio from 1.21.0 to 1.21.1 (#5150)
- build(deps): bump semver from 1.0.13 to 1.0.14 (#5176)
- Fix typos in release-process.md (#5181)
- build(deps): bump thiserror from 1.0.34 to 1.0.35 (#5160)
- build(deps): bump tj-actions/changed-files from 29.0.6 to 29.0.7 (#5159)
- build(deps): bump itertools from 0.10.3 to 0.10.4 (#5149)
- build(deps): bump tj-actions/changed-files from 29.0.4 to 29.0.6 (#5148)
- build(deps): bump criterion from 0.3.6 to 0.4.0 (#5136)
- build(deps): bump serde\_with from 2.0.0 to 2.0.1 (#5135)
- build(deps): bump tj-actions/changed-files from 29.0.3 to 29.0.4 (#5117)
- build(deps): bump tokio-util from 0.7.3 to 0.7.4 (#5118)
- build(deps): bump prost, tonic, tonic-build and console-subscriber (#5009)
- build(deps): bump thiserror from 1.0.33 to 1.0.34 (#5074)
- build(deps): bump tj-actions/changed-files from 29.0.2 to 29.0.3 (#5073)
- change(deps): bump rocksdb from 0.18.0 to 0.19.0 (#5071)
- build(deps): bump owo-colors from 3.4.0 to 3.5.0 (#5037)
- build(deps): bump pin-project from 1.0.11 to 1.0.12 (#4914)
- build(deps): bump futures from 0.3.23 to 0.3.24 (#5020)
- build(deps): bump once\_cell from 1.13.0 to 1.14.0 (#5057)
- build(deps): bump tokio from 1.20.1 to 1.21.0 (#5058)
- build(deps): bump futures-util from 0.3.23 to 0.3.24 (#5047)
- build(deps): bump chrono from 0.4.20 to 0.4.22 (#4984)
- build(deps): bump serde\_json from 1.0.83 to 1.0.85 (#5021)
- build(deps): bump futures-core from 0.3.23 to 0.3.24 (#4998)
- fix(clippy): fix a new nightly Rust clippy "unwrap\_or\_else uses default" lint (#5131)
- fix(clippy): Resolve unnecessary borrow lint and remove redundant imports (#5045)
- ci(build): re-enable Rust beta tests in ubuntu (#5024)
- Document how to resolve disk full errors in CI (#5105)
- Explain how Mergify works, and how it can be used by developers (#5100)
- Explain how to find errors in CI (#5080)
- fix(doc): Continious integration book doc (#5038)
- change(doc): Make the release instructions more explicit (#5026)
- build(deps): bump futures-util from 0.3.21 to 0.3.23 (#4916)
- build(deps): bump vergen from 7.3.2 to 7.4.2 (#5022)
- build(deps): bump google-github-actions/auth from 0.8.0 to 0.8.1 (#5029)
- build(deps): bump bech32 from 0.9.0 to 0.9.1 (#4912)
- deps(build): Bump insta to 1.19.1 (#5018)
- build(deps): bump thiserror from 1.0.32 to 1.0.33 (#5019)
- doc(ci): Explain how to resolve duplicate dependency warnings (#5017)

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,146 @@ All notable changes to Zebra are documented in this file.
...
+- Delete cached state images older than 2 days, but keep a few recent images
+  (#5113, #5124, #5082, #5079)
+- Simplify GitHub actions caches (#5104)
+- Use 200GB disks for managed instances (#5084)
+- Improve test reliability and test output readability (#5014)
+
+#### Zebra Dependencies
+
+- Breaking: Update prost, tonic, tonic-build and console-subscriber to the latest versions (#5009)
+- change(deps): upgrade `zcash\_script` and other shared zcash dependencies (#4926)
```
Comment (upbqdn) [inline CHANGELOG.md]:
```suggestion
- Upgrade `zcash\_script` and other shared zcash dependencies (#4926)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,7 +4,146 @@ All notable changes to Zebra are documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org).
 
-## [Zebra 1.0.0-beta.14](https://github.com/ZcashFoundation/zebra/releases/tag/v1.0.0-beta.13) - 2022-08-30
+## [Zebra 1.0.0-beta.15](https://github.com/ZcashFoundation/zebra/releases/tag/v1.0.0-beta.15) - 2022-09-20
+
+This release improves Zebra's sync and RPC performance, improves test coverate and reliability, and starts creating Docker Hub binaries for releases.
```
Comment (upbqdn) [inline CHANGELOG.md]:
```suggestion
This release improves Zebra's sync and RPC performance, improves test coverage and reliability, and starts creating Docker Hub binaries for releases.
```

**Review by upbqdn — CHANGES_REQUESTED** [review]
All looks good. I'm requesting changes just because of two small typos.

**Review by upbqdn — DISMISSED** [review]

**Comment by upbqdn** [comment]
I realized it's easier to approve the PR since we don't merge PRs until all conversations are resolved anyway. (It's funny because I've been doing it this way for ages. I don't know what I was thinking when I was requesting the changes.)

**Review by upbqdn — APPROVED** [review]

**Review by dconnolly — APPROVED** [review]
Reviewed and tested `cargo install --locked --git https://github.com/ZcashFoundation/zebra --branch beta-15-release zebra` on macOS 12.6 on an M1, installed and ran successfully

**Pull request merged**